packages/hardhat-core/src/internal/hardhat-network/stack-traces/solidity-errors.ts

```diff
@@ -271,7 +271,8 @@ function getMessageFromLastStackTraceEntry(
       }
 
       if (!stackTraceEntry.message.isEmpty()) {
-        return `VM Exception while processing transaction: reverted with an unrecognized custom error`;
+        const selector = stackTraceEntry.message.getSelector() ?? "";
```

Comment:
I think in most cases the selector will exist, but if for some reason it doesn't, the error message will say "with an unrecognized custom error with selector ", which would be ugly and confusing. So I would do something like this:

```suggestion
        const selector = stackTraceEntry.message.getSelector() ?? "<unknown selector>";
```

It's likely that we could do something better here: if there is some return data but it's shorter than 4 bytes, something weird is going on and we should reflect that. But for the time being, this should be enough.

Reply:
Thank you for the suggestion
I use a `UNKNOWN_ERROR_SELECTOR` as unknown selector 
Please check 🙏🏻 [15e92d9](https://github.com/NomicFoundation/hardhat/pull/3384/commits/15e92d9c10e9f60f8f45e2a3a3cfd01cdee6d46c)